webapp/Connector/src/utility/PlotTooltip.js

```diff
@@ -0,0 +1,471 @@
+/*
+ * Copyright (c) 2015-2016 LabKey Corporation
+ *
+ * Licensed under the Apache License, Version 2.0: http://www.apache.org/licenses/LICENSE-2.0
+ */
+// This is a singleton for charting utilities
+Ext.define('Connector.utility.PlotTooltip', {
+
+    alternateClassName: ['PlotTooltipUtils'],
+
+    singleton: true,
+
+    loadBinTooltipData: function(plot, point, datas, plotName)
+    {
+        var rawRows = this.getRawDataRows(plot, datas);
+        var participantSeqContainerMap = {};
+        Ext.each(rawRows, function(row){
+            var participantSeq = row[QueryUtils.SUBJECT_SEQNUM_ALIAS];
+            participantSeqContainerMap[participantSeq] = row[QueryUtils.CONTAINER_ALIAS];
+        });
+        var participantSeqs = Object.keys(participantSeqContainerMap);
+
+        // in the case when a bin contains > 200 participant, skip querying studies for better perf
+        if (participantSeqs.length < 200)
+        {
+            LABKEY.Query.selectRows({
+                schemaName: 'cds',
+                queryName: 'gridbase',
+                method: 'POST',
+                filterArray: [
+                    LABKEY.Filter.create('ParticipantSequenceNum', Object.keys(participantSeqContainerMap).join(';'), LABKEY.Filter.Types.IN)
+                ],
+                success: function (results) {
+                    var records = results.rows, filteredRecords = [];
+                    Ext.each(records, function(record){
+                        if (participantSeqContainerMap[record.ParticipantSequenceNum] == record.Container)
+                            filteredRecords.push(record);
+                    });
+                    var plotData = {
+                        rows: filteredRecords
```

Comment:
Is plotData.rows ever used? I don't believe it is, but if so, this should probably also be added to line 50 below so that it sends the following as the last param:

{rows: [], studyLabels: []}